### PR TITLE
Added `PoolSwap`  RPC 

### LINKED ISSUE #48: DeFiChain Jellyfish RPC Implementations
> This tracks all the RPC methods and categories being implemented in `jellyfish-core`. Will list all the methods here, and strike out those that are not applicable.

### `blockchain.cpp`
- [ ] getblockchaininfo
- [ ] getchaintxstats
- [ ] getblockstats
- [ ] getbestblockhash
- [ ] getblockcount
- [ ] getblock
- [ ] getblockhash
- [ ] getblockheader
- [ ] getchaintips
- [ ] getdifficulty
- [ ] getmempoolancestors
- [ ] getmempooldescendants
- [ ] getmempoolentry
- [ ] getmempoolinfo
- [ ] clearmempool
- [ ] getrawmempool
- [ ] gettxout
- [ ] gettxoutsetinfo
- [ ] pruneblockchain
- [ ] savemempool
- [ ] verifychain
- [ ] preciousblock
- [ ] scantxoutset
- [ ] getblockfilter
- [ ] invalidateblock
- [ ] reconsiderblock
- [ ] waitfornewblock
- [ ] waitforblock
- [ ] waitforblockheight
- [ ] syncwithvalidationinterfacequeue

### `mining.ccp`
* [x] getnetworkhashps
* [x] getmintinginfo
* [ ] prioritisetransaction
* [ ] getblocktemplate
* [ ] submitblock
* [ ] submitheader
* [ ] generatetoaddress
* [ ] estimatesmartfee
* [ ] estimaterawfee

### `misc.cpp`
- [ ] getmemoryinfo
- [ ] logging
- [ ] validateaddress
- [ ] createmultisig
- [ ] deriveaddresses
- [ ] getdescriptorinfo
- [ ] verifymessage
- [ ] signmessagewithprivkey
- [ ] setmocktime
- [ ] echo
- [ ] echojson

### `net.cpp`
- [ ] getconnectioncount
- [ ] ping
- [ ] getpeerinfo
- [ ] addnode
- [ ] disconnectnode
- [ ] getaddednodeinfo
- [ ] getnettotals
- [ ] getnetworkinfo
- [ ] setban
- [ ] listbanned
- [ ] clearbanned
- [ ] setnetworkactive
- [ ] getnodeaddresses

### `rawtransaction.cpp`
- [ ] getrawtransaction
- [ ] createrawtransaction
- [ ] decoderawtransaction
- [ ] decodescript
- [ ] sendrawtransaction
- [ ] combinerawtransaction
- [ ] signrawtransactionwithkey
- [ ] testmempoolaccept
- [ ] decodepsbt
- [ ] combinepsbt
- [ ] finalizepsbt
- [ ] createpsbt
- [ ] converttopsbt
- [ ] utxoupdatepsbt
- [ ] analyzepsbt
- [ ] joinpsbts
- [ ] gettxoutproof
- [ ] verifytxoutproof

### `server.cpp`
- [ ] getrpcinfo
- [ ] help
- [ ] stop
- [ ] uptime

### `spv_rpc.cpp`
- [ ] spv_sendrawtx
- [ ] spv_createanchor
- [ ] spv_createanchortemplate
- [ ] spv_estimateanchorcost
- [ ] spv_rescan
- [ ] spv_syncstatus
- [ ] spv_gettxconfirmations
- [ ] spv_splitutxo
- [ ] spv_listanchors
- [ ] spv_listanchorauths
- [ ] spv_listanchorrewardconfirms
- [ ] spv_listanchorrewards
- [ ] spv_listanchorsunrewarded
- [ ] spv_listanchorspending
- [ ] spv_setlastheight

### `rpcwallet.cpp`
- [ ] fundrawtransaction
- [ ] abandontransaction
- [ ] abortrescan
- [ ] addmultisigaddress
- [ ] backupwallet
- [ ] bumpfee
- [ ] createwallet
- [ ] dumpprivkey
- [ ] dumpwallet
- [ ] encryptwallet
- [ ] getaddressesbylabel
- [ ] getaddressinfo
- [x] getbalance
- [ ] getnewaddress
- [ ] getrawchangeaddress
- [ ] getreceivedbyaddress
- [ ] getreceivedbylabel
- [ ] gettransaction
- [ ] getunconfirmedbalance
- [ ] getbalances
- [ ] getwalletinfo
- [ ] importaddress
- [ ] importmulti
- [ ] importprivkey
- [ ] importprunedfunds
- [ ] importpubkey
- [ ] importwallet
- [ ] keypoolrefill
- [ ] listaddressgroupings
- [ ] listlabels
- [ ] listlockunspent
- [ ] listreceivedbyaddress
- [ ] listreceivedbylabel
- [ ] listsinceblock
- [ ] listtransactions
- [ ] listunspent
- [ ] listwalletdir
- [ ] listwallets
- [ ] loadwallet
- [ ] lockunspent
- [ ] removeprunedfunds
- [ ] rescanblockchain
- [ ] sendmany
- [ ] sendtoaddress
- [ ] sethdseed
- [ ] setlabel
- [ ] settxfee
- [ ] setwalletflag
- [ ] signmessage
- [ ] signrawtransactionwithwallet
- [ ] unloadwallet
- [ ] walletcreatefundedpsbt
- [ ] walletlock
- [ ] walletpassphrase
- [ ] walletpassphrasechange
- [ ] walletprocesspsbt

### `mn_rpc.cpp`
- [ ] setgov
- [ ] getgov
- [ ] isappliedcustomtx

### `rpc_accounts.cpp`
- [ ] listaccounts
- [ ] getaccount
- [ ] gettokenbalances
- [ ] utxostoaccount
- [ ] accounttoaccount
- [ ] accounttoutxos
- [ ] listaccounthistory
- [ ] accounthistorycount
- [ ] listcommunitybalances
- [ ] sendtokenstoaddress

### `rpc_masternodes.cpp`
- [ ] createmasternode
- [ ] resignmasternode
- [ ] listmasternodes
- [ ] getmasternode
- [ ] getmasternodeblocks
- [ ] listcriminalproofs
- [ ] getanchorteams
- [ ] getactivemasternodecount
- [ ] listanchors

### `rpc_poolpair.cpp`
- [ ] listpoolpairs
- [ ] getpoolpair
- [ ] addpoolliquidity
- [ ] removepoolliquidity
- [ ] createpoolpair
- [ ] updatepoolpair
- [ ] poolswap
- [ ] listpoolshares
- [ ] testpoolswap

### `rpc_tokens.cpp`
- [ ] createtoken
- [ ] updatetoken
- [ ] listtokens
- [ ] gettoken
- [ ] getcustomtx
- [ ] minttokens

### PR DESCRIPTION
**Pull request by siradji**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

RPC implementation #48 

#### Which issue(s) does this PR fixes?:

Fixes part of #48

### CODE REVIEW

File: packages/jellyfish-api-core/src/category/poolpair.ts [hunk 1]
```diff
@@ -104,6 +104,25 @@ export class PoolPair {
...
+   *  @param {string} metadata.to address of the owner of tokenTo
+   *  @param {string} metadata.tokenTo swap to token {symbol/id}
+   *  @param {number} [metadata.maxPrice] The aximum acceptable price
+   *  @param {PoolSwapUTXO[]} [UTXOS = []] Array for utxos to spend from.
+   *  @param {string} [UTXOS.txid] The transaction id.
+   *  @param {number} [UTXOS.vout] The output number.
+   *  @return {Promise<string>}  Hex of performed transaction
+   */
+  async poolSwap (metadata: PoolSwapMetadata, UTXOS: PoolSwapUTXO[] = []): Promise<string> {
+    return await this.client.call('poolswap', [metadata, UTXOS], 'bignumber')
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/poolpair.ts:124]:
```suggestion
  async poolSwap (metadata: PoolSwapMetadata, utxos: PoolSwapUTXO[] = []): Promise<string> {
    return await this.client.call('poolswap', [metadata, utxos], 'bignumber')
```

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const poolpairResultAfter = Object.values(await container.call('getpoolpair', ['CAT-DFI']))[0] as PoolPairInfo
+    const reserveAAfter = new BigNumber(poolpairResultBefore.reserveB).plus(150)
+    const reserveBAfter = new BigNumber(poolpairResultBefore.totalLiquidity).pow(2).div(reserveAAfter) // sqrt(a * b)^2/b
+
+    expect(poolpairResultAfter.reserveB.toFixed(0)).toStrictEqual(reserveAAfter.toFixed(0))
+    expect(poolpairResultAfter.reserveA.toFixed(0)).toStrictEqual(reserveBAfter.toFixed(0))
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:112]: do you need `toFixed(0)` here??

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should poolSwap with UTXOS ', async () => {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:29]:
```suggestion
  it('should poolSwap with utxos ', async () => {
```

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    const tokenAddress = await getNewAddress(container)
+    const dfiAddress = await getNewAddress(container)
+    const poolLiquidityAddress = await getNewAddress(container)
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:32]: you can use `getaccount` to check the after poolswap balance changes, its a guidance for people know how to manage their coin in each account

**Review by canonbrother — COMMENTED** [review]
test ordering
from simple -> complicated
eg:
1. should poolSwap
2. should poolSwap with utxos

**Comment by fuxingloh** [comment]
> #### What kind of PR is this?:
> /kind feature
> 
> #### What this PR does / why we need it:
> RPC implementation #48
> 
> #### Which issue(s) does this PR fixes?:
> Fixes #48
> 
> #### Additional comments?:

Edit your comment body to "Fixes part of #48" or it will close that entire issue.

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts (hunk 4 above)
Comment (siradji) [reply packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:33]: I have added the accountReceiver balance check to `poolSwap` block. I think that will suffice.

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+    const hex = await client.poolpair.poolSwap(metadata)
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const poolpairResultAfter = Object.values(await container.call('getpoolpair', ['CAT-DFI']))[0] as PoolPairInfo
+    const reserveBAfter = new BigNumber(poolpairResultBefore.reserveB).plus(555)
+    const reserveAAfter = new BigNumber(poolpairResultBefore.totalLiquidity).pow(2).div(reserveBAfter) // sqrt(a * b)^2/b
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:66]: what is this? `sqrt(a * b)^2/b`

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+    expect(poolpairResultAfter.reserveB.toFixed(0)).toStrictEqual(reserveBAfter.toFixed(0))
+    expect(poolpairResultAfter.reserveA.toFixed(0)).toStrictEqual(reserveAAfter.toFixed(0))
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:69]: does `reserveB` need `toFixed()`?
please use the `toFixed()` properly
and please add the exact result amount at the end of code, easier for reviewer
```
...code.... // {amount}
```

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    const utxos = (await container.call('listunspent'))
+      .filter((utxos: any) => utxos.txid === txid)
+      .map((utxos: any) => {
+        return {
+          txid: utxos.txid,
+          vout: utxos.vout
+        }
+      })
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:113]:
```suggestion
    const utxos = (await container.call('listunspent'))
      .filter((utxo: any) => utxo.txid === txid)
      .map((utxo: any) => {
        return {
          txid: utxo.txid,
          vout: utxo.vout
        }
      })
```

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+  it('should not poolSwap with invalid token', async () => {
+    const tokenAddress = await getNewAddress(container)
+    const dfiAddress = await getNewAddress(container)
+    const poolLiquidityAddress = await getNewAddress(container)
+
+    await createToken(container, 'FOX', { collateralAddress: tokenAddress })
+    await utxosToAccount(container, 500, { address: dfiAddress })
+    await mintTokens(container, 'FOX', { address: dfiAddress })
+    await createPoolPair(container, 'FOX', 'DFI', { status: false })
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:217]: why `status` false here??

Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:217]: do you need the those fixtures??

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+      shareAddress: poolLiquidityAddress
+    })
+
+    const metadata: PoolSwapMetadata = {
+      from: dfiAddress,
+      tokenFrom: 'DFI',
+      amountFrom: 492,
+      to: await getNewAddress(container),
+      tokenTo: 'ETH',
+      maxPrice: 10
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:103]: the test case is `should poolSwap with utxos`
why `maxPrice` here??

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+    expect(poolpairResultAfter.reserveA.toFixed(0)).toStrictEqual(reserveAAfter.toFixed(0))
+
+    const accountReceiver = (await client.account.getAccount(addressReceiver))[0]
+    const accountReceiverBalance = new BigNumber(accountReceiver.split('@')[0])
+    const amountReceived = new BigNumber(poolpairResultBefore.reserveA).minus(reserveAAfter)
+
+    expect(accountReceiverBalance.toFixed(0)).toStrictEqual(amountReceived.toFixed(0))
+  })
+
+  it('should poolSwap with utxos ', async () => {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:78]:
```suggestion
  it('should poolSwap with utxos', async () => {
```

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+    await createPoolPair(container, 'DOG', 'DFI', { status: false })
+    await addPoolLiquidity(container, {
+      tokenA: 'DOG',
+      amountA: 1000,
+      tokenB: 'DFI',
+      amountB: 500,
+      shareAddress: poolLiquidityAddress
+
+    })
+    const metadata: PoolSwapMetadata = {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:199]: break line

**Review by canonbrother — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts (hunk 5 above)
Comment (siradji) [reply packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:66]: Just stating what happened there. Can remove it.

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts (hunk 6 above)
Comment (siradji) [reply packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:69]: Got it.

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts (hunk 8 above)
Comment (siradji) [reply packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:217]: No we don't and have been removed

File: packages/jellyfish-api-core/src/category/poolpair.ts [hunk 12]
```diff
@@ -198,6 +217,20 @@ export interface PoolShareOptions {
...
+export interface PoolSwapMetadata {
+  from: string
+  tokenFrom: string
+  amountFrom: number
+  to: string
+  tokenTo: string
+  maxPrice?: number
+}
+
+export interface PoolSwapUTXO {
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/poolpair.ts:229]: Call this `UTXO` and rename everything in this file that was called `*UTXO` to use that single interface.

Similar to this https://github.com/DeFiCh/jellyfish/blob/015ec69bf41c73376ff1f659645e0095aceb7a42/packages/jellyfish-api-core/src/category/oracle.ts#L177-L180

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/src/category/poolpair.ts [hunk 13]
```diff
@@ -105,6 +105,25 @@ export class PoolPair {
...
+   * @param {string} metadata.from address of the owner of tokenFrom
+   * @param {string} metadata.tokenFrom swap from token {symbol/id}
+   * @param {number} metadata.amountFrom amount from tokenA
+   * @param {to} metadata.to address of the owner of tokenTo
+   * @param {tokenTo} metadata.tokenTo swap to token {symbol/id}
+   * @param {maxPrice} [metadata.maxPrice] acceptable max price
+   *  @param {PoolSwapUTXO[]} [utxos = []] array for utxos to spend from.
+   *  @param {string} [utxos.txid] the transaction id.
+   *  @param {number} [utxos.vout] the output number.
+   *  @return {Promise<string>}  hex of performed transaction
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/poolpair.ts:121]: alignment

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    const utxos = (await container.call('listunspent'))
+      .filter((utxo: any) => utxo.txid === txid)
+      .map((utxo: any) => {
+        return {
+          txid: utxo.txid,
+          vout: utxo.vout
+        }
+      })
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:113]: @siradji 
that was my bad pattern on testing utxo input
can you follow this??
```ts
await container.fundAddress(poolLiquidityAddress, 25)
const utxos = await container.call('listunspent')
const utxo = utxos.find((utxo: any) => utxo.address === poolLiquidityAddress)

const rawtx = await container.call('getrawtransaction', [txid, true])
expect(rawtx.vin[0].txid).toStrictEqual(utxo.txid)
```

File: website/docs/jellyfish/api/poolpair.md [hunk 15]
```diff
@@ -161,6 +161,29 @@ interface PoolShareOptions {
 }
 ```
 
+## poolSwap 
+Creates a poolSwap transaction.
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/poolpair.md:165]: `Creates a pool swap transaction with given metadata` Like what you wrote in another file

File: website/docs/jellyfish/api/poolpair.md [hunk 16]
```diff
@@ -161,6 +161,29 @@ interface PoolShareOptions {
...
+interface PoolSwapMetadata {
+  from: string
+  tokenFrom: string
+  amountFrom: number
+  to: string
+  tokenTo: string
+  maxPrice?: number
+}
+
+interface PoolSwapUTXO {
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/poolpair.md:181]: Change this interface
(Refer to https://github.com/DeFiCh/jellyfish/pull/413/files#r668418832)

**Comment by jingyi2811** [comment]
Screenshot 2021-07-14 at 5 41 27 PM

Both interfaces have the same variables and are written in the **same file**. Hence,

You should remove `TestPoolSwapMetadata`

Both rpcs should share `PoolSwapMetadata`

**Comment by jingyi2811** [comment]
You need to resolve the merge conflict too

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const poolpairResultAfter = Object.values(await container.call('getpoolpair', ['CAT-DFI']))[0] as PoolPairInfo
+    const reserveBAfter = new BigNumber(poolpairResultBefore.reserveB).plus(555) // 1055
+    const reserveAAfter = new BigNumber(poolpairResultBefore.totalLiquidity).pow(2).div(reserveBAfter) // 473.93364928032265610654
+
+    expect(new BigNumber(poolpairResultAfter.reserveB)).toStrictEqual(reserveBAfter)
+    expect(poolpairResultAfter.reserveA.toFixed(0)).toStrictEqual(reserveAAfter.toFixed(0))
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:69]:
```suggestion
    expect(poolpairResultAfter.reserveA.toFixed(8)).toStrictEqual(reserveAAfter.toFixed(8))
```

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+
+    const accountReceiver = (await client.account.getAccount(addressReceiver))[0]
+    const accountReceiverBalance = new BigNumber(accountReceiver.split('@')[0]) // 526.06635072
+    const amountReceived = new BigNumber(poolpairResultBefore.reserveA).minus(reserveAAfter) // 526.06635071967734389346
+
+    expect(accountReceiverBalance.toFixed(0)).toStrictEqual(amountReceived.toFixed(0))
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:75]:
```suggestion
    expect(accountReceiverBalance.toFixed(8)).toStrictEqual(amountReceived.toFixed(8))
```

**Comment by jingyi2811** [comment]
I think you accidentally commit and push `husky.sh`

File: packages/jellyfish-api-core/src/category/poolpair.ts [hunk 19]
```diff
@@ -27,7 +27,7 @@ export class PoolPair {
    * @param {number} utxos.vout
    * @return {Promise<string>}
    */
-  async createPoolPair (metadata: CreatePoolPairMetadata, utxos: CreatePoolPairUTXO[] = []): Promise<string> {
+  async createPoolPair (metadata: CreatePoolPairMetadata, utxos: UTXO[] = []): Promise<string> {
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/poolpair.ts:30]: Remove `CreatePoolPairUTXO` from the js comment too

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 20]
```diff
@@ -0,0 +1,219 @@
...
+    await utxosToAccount(container, 1000, { address: dfiAddress })
+    await mintTokens(container, 'DOG', { address: dfiAddress })
+    await createPoolPair(container, 'DOG', 'DFI', { status: false })
+    await addPoolLiquidity(container, {
+      tokenA: 'DOG',
+      amountA: 1000,
+      tokenB: 'DFI',
+      amountB: 500,
+      shareAddress: poolLiquidityAddress
+
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts:195]: Remove break line

File: packages/jellyfish-api-core/src/category/poolpair.ts [hunk 21]
```diff
@@ -202,19 +221,14 @@ export interface AddPoolLiquiditySource {
 }
 
 export interface AddPoolLiquidityOptions {
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/poolpair.ts:223]: `AddPoolLiquidityOptions` and `RemovePoolLiquidityOptions` interfaces have same variable. They should be `PoolLiquidityOptions`

**Review by canonbrother — DISMISSED** [review]

**Review by jingyi2811 — APPROVED** [review]

File: website/docs/jellyfish/api/poolpair.md [hunk 22]
```diff
@@ -104,18 +104,18 @@ Add pool liquidity transaction
 
 ```ts title="client.poolpair.addPoolLiquidity()"
 interface poolpair {
-  addPoolLiquidity (from: AddPoolLiquiditySource, shareAddress: string, options: AddPoolLiquidityOptions = {}): Promise<string>
+  addPoolLiquidity (from: PoolLiquidityOptions, shareAddress: string, options: PoolLiquidityOptions = {}): Promise<string>
```
Comment (fuxingloh) [inline website/docs/jellyfish/api/poolpair.md:107]:
```suggestion
  addPoolLiquidity (from: AddPoolLiquiditySource, shareAddress: string, options: PoolLiquidityOptions = {}): Promise<string>
```

**Review by fuxingloh — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Review by jingyi2811 — APPROVED** [review]

**Pull request merged**